lib/rgb_matrix/animation.ex

```diff
@@ -1,109 +1,88 @@
 defmodule RGBMatrix.Animation do
-  @moduledoc """
-  Provides a data structure and functions to define an RGBMatrix animation.
+  alias Layout.LED
 
-  There are currently two distinct ways to define an animation.
+  @callback new(leds :: list(LED.t()), config :: any) :: {render_in, any}
+  @callback render(state :: any, config :: any) ::
+              {list(RGBMatrix.any_color_model()), render_in, any}
+  @callback interact(state :: any, config :: any, led :: LED.t()) :: {render_in, any}
```

Comment:
No, the `state` here in the behaviour callback definition is the black box defined by each individual animation.  You can't assume what they are going to use as state, some might use structs, some might use maps, some might just need a single integer, who knows.  `RGBMatrix.render/1` and `RGBMatrix.interact/2` DO use `t` in their typespecs.  The bahviour cannot, because it's referring to (for example) `RGBMatrix.Animation.HueWave.render/1`.

A follow-up suggestion you might make then is that each animation should instead of receiving the black box state as input, should receive the animation struct as input.  Maybe that's a good idea, but let's discuss that separately...